src/db.c

```diff
@@ -297,7 +298,10 @@ static void dbSetValue(serverDb *db, robj *key, robj *val, int overwrite, dictEn
         old = dictGetVal(de);
     }
     kvstoreDictSetVal(db->keys, slot, de, val);
-    if (server.lazyfree_lazy_server_del) {
+    /* For efficiency, let the I/O thread that allocated an object also deallocate it. */
+    if (tryOffloadFreeObjToIOThreads(old) == C_OK) {
+        return;
```

Comment:
Returns here has no effect but it adds some mental cost to read this code. We don't do return in the `else if` and `else`, so better not do it here either.

```suggestion
        /* OK */
```

Reply:
Changed